configuration/pih/htmlforms/checkin.xml

```diff
@@ -185,10 +188,14 @@
                     </h3>
 
                     <p class="left">
-                        <obs conceptId="PIH:Receipt number" id="receiptNumber" showUnits="false"
+                        <obs value="number" conceptId="PIH:Receipt number" id="receiptNumber" showUnits="false"
```

Comment:
What does `value="number"` do?  This doesn't seem right, we don't want to set the value of the data to the text of "number".

Reply:
`style="dropdown"` doesn't really make sense here does it?  This is a text box.  Also `unitsCssClass` isn't really needed, since we have `showUnits="false"`, right?

Reply:
You are Right

Reply:
This concept is text (class = question; datatype = text).  Only datatype of numeric have units (ie. weight (kg) or blood volume (Liter).

![image](https://github.com/PIH/openmrs-config-zl/assets/1164849/66ca946d-b9d1-4dee-8c8c-9a2f089f0afe)

As @mseaton suggested, style=dropdown is appropriate for datatype=coded where there are answers to a question (ie. Color of automobile = red, white, orange)

Reply:
According to this Ticket https://pihemr.atlassian.net/browse/HAI-754
I had to allow only numeric value to the input
Before the change, the type was text as you say
this might be an error  in the ticket, I quess

Reply:
Your approach seems correct @Dutervil .  I don't think we need or want a new Concept, just want to restrict the input box to a "numeric", though this might have an adverse impact on editing existing data.  By using the existing Text Concept, you will get a text box that is not a number field, but your change to use JS to make this a number input seems reasonable.

Reply:
Thank for your approach @mseaton